### ISSUE TITLE
Frequent "no method named `as_encoded_bytes`" error

### ISSUE THREAD

**Issue by max-sixty** [willcrichton/flowistry#98]
# Problem

I get this error I think every time I launch VS Code. Is it possibly a version mismatch, and installing with `--locked` would solve it?

Thank you!

# Logs

**OS:** darwin (23.5.0)
**VSCode:** 1.91.1
**Error message**
```
    Updating crates.io index
  Installing flowistry_ide v0.5.41
    Updating crates.io index
   Compiling proc-macro2 v1.0.86
   Compiling unicode-ident v1.0.12
   Compiling serde v1.0.204
   Compiling equivalent v1.0.1
   Compiling hashbrown v0.14.5
   Compiling winnow v0.5.40
   Compiling camino v1.1.7
   Compiling log v0.4.22
   Compiling cfg-if v1.0.0
   Compiling byteorder v1.5.0
   Compiling serde_json v1.0.121
   Compiling anyhow v1.0.86
   Compiling semver v1.0.23
   Compiling index_vec v0.1.3
   Compiling splitmut v0.2.1
   Compiling ryu v1.0.18
   Compiling memchr v2.7.4
   Compiling smallvec v1.13.2
   Compiling itoa v1.0.11
   Compiling fxhash v0.2.1
   Compiling clap_lex v0.7.2
   Compiling adler v1.0.2
   Compiling intervaltree v0.2.7
error[E0599]: no method named `as_encoded_bytes` found for reference `&OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:187:26
    |
187 |         let bytes = self.as_encoded_bytes();
    |                          ^^^^^^^^^^^^^^^^ method not found in `&OsStr`

error[E0599]: no method named `as_encoded_bytes` found for reference `&OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:196:26
    |
196 |         let bytes = self.as_encoded_bytes();
    |                          ^^^^^^^^^^^^^^^^ method not found in `&OsStr`

   Compiling heck v0.5.0
error[E0599]: no method named `as_encoded_bytes` found for reference `&OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:202:26
    |
202 |         let bytes = self.as_encoded_bytes();
    |                          ^^^^^^^^^^^^^^^^ method not found in `&OsStr`

error[E0599]: no function or associated item named `from_encoded_bytes_unchecked` found for struct `OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:207:29
    |
207 |             unsafe { OsStr::from_encoded_bytes_unchecked(s) }
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `OsStr`

error[E0599]: no method named `as_encoded_bytes` found for reference `&OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:211:26
    |
211 |         let bytes = self.as_encoded_bytes();
    |                          ^^^^^^^^^^^^^^^^ method not found in `&OsStr`

error[E0599]: no method named `as_encoded_bytes` found for reference `&OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:226:29
    |
226 |         let haystack = self.as_encoded_bytes();
    |                             ^^^^^^^^^^^^^^^^ method not found in `&OsStr`

error[E0599]: no function or associated item named `from_encoded_bytes_unchecked` found for struct `OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:234:24
    |
234 |                 OsStr::from_encoded_bytes_unchecked(first),
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `OsStr`

error[E0599]: no function or associated item named `from_encoded_bytes_unchecked` found for struct `OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:235:24
    |
235 |                 OsStr::from_encoded_bytes_unchecked(second),
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `OsStr`

error[E0599]: no method named `as_encoded_bytes` found for reference `&OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:277:24
    |
277 |         let bytes = os.as_encoded_bytes();
    |                        ^^^^^^^^^^^^^^^^ method not found in `&OsStr`

error[E0599]: no function or associated item named `from_encoded_bytes_unchecked` found for struct `OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:280:20
    |
280 |             OsStr::from_encoded_bytes_unchecked(first),
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `OsStr`

error[E0599]: no function or associated item named `from_encoded_bytes_unchecked` found for struct `OsStr` in the current scope
   --> /Users/maximilian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_lex-0.7.2/src/ext.rs:281:20
    |
281 |             OsStr::from_encoded_bytes_unchecked(second),
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `OsStr`

   Compiling anstyle v1.0.8
For more information about this error, try `rustc --explain E0599`.
error: could not compile `clap_lex` (lib) due to 11 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `flowistry_ide v0.5.41`, intermediate artifacts can be found at `/var/folders/wf/s6ycxvvs4ln8qsdbfx40hnc40000gn/T/cargo-install7DNtCm`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

```

**Full log:** https://paste.rs/yc1dt

**Comment by willcrichton** [comment]
Ah yes I was missing a `--locked` on the `cargo install`. I just released v0.5.42 which fixes this issue, and your VSCode should automatically update. Please let me know if the error persists.}

**Comment by max-sixty** [comment]
Thanks! That fixed that one :)
